### ISSUE TITLE
Docker image

### ISSUE THREAD

**Issue by marcdumais-work** [eclipse-theia/theia-blueprint#75]
### Feature Description:

Before Theia joined the EF, we had the [theia-apps repo](https://github.com/theia-ide/theia-apps). Because  of its popularity and diversity of contributors, it's a hard repo to re-license, making it difficult to bring under the Foundation. 

However, with the move to vscode extensions to support programming languages, it's becoming less important to have multiple, per-language examples, when the main difference is which extensions are bundled. They could essentially all use the same Theia application, in theory.  

I think the theia-apps examples can remain, with minimal maintenance, for a while. But we can also make one or a couples, based on our experiences with theia-apps and the lessons learned there. 

Let's start with a modest image, a Blueprint-based IDE appliance, for JavaScript/TypeScript software development. 

Let's re-use the browser package we'll make in #74

**Comment by AlexTalker** [comment]
I'd love to see docker support, recently I tried to make up package for `theia-full` from the apps repo but that try ended up in some bugs in the end and I wanted to check if docker image works alright but it just won't download and seems to not be supported anymore.
So I would really like example of some docker-based stuff that actually works based on `theia`.
Then I might bodge up example of how to package this stuff for Linux distros, as now i only see the desktop version everywhere but I mostly access stuff remotely and `VNC/RDP` just doesn't cut it for me, I'd love to have something browser-based on the other side.

Sorry for the occasional ramble, just wanted to highlight that docker-based version would be helpful to at least some people.

**Comment by dginhoux** [comment]
Because the last official docker image theia1.17 (and old releases tool) was removed from docker hub, i know a corruption problem on my local registry and not be able to pull theses old theia image.

I've found an alternative here : https://hub.docker.com/r/gitpod/openvscode-server

**Comment by JonasHelming** [comment]
Good news: The default container from the Theia app repo is published here by a contributor:
https://quay.io/repository/zowe-explorer/theia?tab=tags

**Comment by czadikem** [comment]
This still being worked on?

**Comment by jfaltermeier** [comment]
Yes, I've started to prepare a Blueprint Docker Image here: https://github.com/eclipsesource/theia-cloud/blob/main/demo/dockerfiles/demo-theia-docker/Dockerfile

This uses this branch at the moment: https://github.com/eclipse-theia/theia-blueprint/tree/jf/docker
This branch does not include the electron Blueprint however. 

Open topics are therefore 
a) integrating the Browser-App from the branch into the main branch so that both electron and browser apps may be built. This might require to split some of the existing packages. 
b) provide a minimal Dockerfile to package this. The Dockerfile linked above e.g. still contains a sample workspace or sets environment variables that should not be set in a Blueprint template
